src/microbe_stage/BackgroundPlane.cs

```diff
@@ -0,0 +1,263 @@
+        backgroundParticles.Rotation = Rotation;
+        backgroundParticles.LocalCoords = false;
+
+        AddChild(backgroundParticles);
+
+        OnDisplayBackgroundParticlesChanged(Settings.Instance.DisplayBackgroundParticles);
+    }
+
+    public void SetWorldPosition(Vector2 position)
+    {
+        currentBackgroundMaterial.SetShaderParameter(worldPositionParameter, position);
+    }
+
+    public void SetVisibility(bool visible)
+    {
+        Visible = visible;
+
+        if (backgroundParticles != null)
+            OnDisplayBackgroundParticlesChanged(Settings.Instance.DisplayBackgroundParticles);
+    }
+
+    public void UpdateLightLevel(float lightLevel)
+    {
+        currentBackgroundMaterial.SetShaderParameter(lightLevelParameter, lightLevel);
+    }
+
+    protected override void Dispose(bool disposing)
+    {
+        if (disposing)
+        {
+            textureAlbedoParameter.Dispose();
+            blurAmountParameter.Dispose();
+            lightLevelParameter.Dispose();
+            distortionStrengthParameter.Dispose();
+            worldPositionParameter.Dispose();
+
+            if (blurPlanePath != null)
+            {
+                blurPlanePath.Dispose();
+                blurColorRectPath.Dispose();
+                backgroundPlanePath.Dispose();
+            }
+        }
+
+        base.Dispose(disposing);
+    }
+
+    private void OnDisplayBackgroundParticlesChanged(bool displayed)
+    {
+        if (backgroundParticles == null)
+        {
+            GD.PrintErr("MicrobeCamera didn't find background particles on settings change");
+            return;
+        }
+
+        backgroundParticles.Emitting = displayed;
+
+        if (displayed)
+        {
+            backgroundParticles.Show();
+        }
+        else
+        {
+            backgroundParticles.Hide();
+        }
+    }
+
+    private void OnBackgroundDistortionChanged(float value)
+    {
+        ApplyDistortionEffect();
+    }
+
+    private void ApplyDistortionEffect()
+    {
+        currentBackgroundMaterial.SetShaderParameter(distortionStrengthParameter,
+            Settings.Instance.MicrobeDistortionStrength.Value);
+    }
+
+    private void OnBackgroundBlurStrengthChanged(float value)
+    {
+        ApplyBlurEffect();
+    }
+
+    private void OnBackgroundBlurToggleChanged(bool value)
+    {
+        ApplyBlurEffect();
+    }
+
+    private void ApplyBlurEffect()
+    {
+        float blurStrength = Settings.Instance.MicrobeBackgroundBlurStrength;
+        bool enabled = blurStrength > 0 && Settings.Instance.MicrobeBackgroundBlurEnabled.Value;
+
+        if (enabled)
+        {
+            blurPlane.Visible = true;
+
+            canvasBlurMaterial.SetShaderParameter(blurAmountParameter, blurStrength);
+            spatialBlurMaterial.SetShaderParameter(blurAmountParameter, blurStrength);
+
+            if (backgroundPlane.GetParent() == this)
```

Comment:
Rather than doing this I think there should be a bool flag to remember what was done to avoid this check.

Reply:
There still wasn't a bool flag added and there is still a GetParent call (which is almost always a sign of too much interconnectness in Godot, so an early sign of spaghetti forming). It's not super serious in this case but I see a very simple way to avoid it here so I'd like that approach to be taken.

Reply:
Would it be a good idea to simply reuse the `blurEnabledLastTime` variable for that, since this hierarchy change is done along with all the other changes associated with `blurEnabledLastTime`?

Reply:
I mean yeah, it probably works to keep it all behind that one variable.